packages/artifacts/src/download/download.node.ts

```diff
@@ -0,0 +1,66 @@
+import { createWriteStream, existsSync } from 'node:fs'
+import { mkdir } from 'node:fs/promises'
+import os from 'node:os'
+import { dirname } from 'node:path'
+import _maybeGetSnarkArtifacts from './download.browser'
+import type { SnarkArtifacts } from './types'
+
+async function download(url: string, outputPath: string) {
+  const response = await fetch(url)
+
+  if (!response.ok)
+    throw new Error(`Failed to fetch ${url}: ${response.statusText}`)
+  if (!response.body) throw new Error('Failed to get response body')
+
+  const dir = dirname(outputPath)
+  await mkdir(dir, { recursive: true })
+
+  const fileStream = createWriteStream(outputPath)
+  const reader = response.body.getReader()
+
+  try {
+    const pump = async () => {
+      const { done, value } = await reader.read()
+      if (done) {
+        fileStream.end()
+        return
+      }
+
+      fileStream.write(Buffer.from(value))
+      await pump()
+    }
+
+    await pump()
+  } catch (error) {
+    fileStream.close()
+    throw error
+  }
+}
+
+// https://unpkg.com/@zk-kit/poseidon-artifacts@latest/poseidon.wasm -> @zk/poseidon-artifacts@latest/poseidon.wasm
+const extractEndPath = (url: string) => url.substring(url.indexOf('@zk'))
+
+async function maybeDownload(url: string) {
+  const outputPath = `${os.tmpdir()}/${extractEndPath(url)}`
+
+  if (!existsSync(outputPath)) await download(url, outputPath)
+
+  return outputPath
+}
+
+export default async function maybeGetSnarkArtifacts(
+  ...pars: Parameters<typeof _maybeGetSnarkArtifacts>
+): Promise<SnarkArtifacts> {
+  const { wasm: wasmUrl, zkey: zkeyUrl } = await _maybeGetSnarkArtifacts(
+    ...pars,
+  )
+  const [wasm, zkey] = await Promise.all([
+    maybeDownload(wasmUrl),
+    maybeDownload(zkeyUrl),
+  ])
+
+  return {
+    wasm,
+    zkey,
+  }
+}
```

Comment:
Different from `@zk-kit/utils`:
Here we were re exporting `Project` and `projects`.
https://github.com/privacy-scaling-explorations/zk-kit/blob/66960e7e037cb657f745e50de30c6843ba20e5a4/packages/utils/src/snark-artifacts/snark-artifacts.node.ts#L73
Why?
is it so that they can be imported when using [this](https://github.com/privacy-scaling-explorations/zk-kit/blob/66960e7e037cb657f745e50de30c6843ba20e5a4/packages/utils/package.json#L52-L60) export?